jaraco/classes/properties.py

```diff
@@ -1,3 +1,22 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, overload
+
+if TYPE_CHECKING:
+    from collections.abc import Callable
+    from typing import Any
+
+    from typing_extensions import NoReturn, Self, TypeAlias
+
+    # TODO(bswck): Migrate to PEP 695 by 2027-10. https://peps.python.org/pep-0695/
```

Comment:
Nice comment: it includes directions on what needs to happen and a hyper link and guidance about timelines.

I'm "-0" on "TODO(name)". While I appreciate claiming responsibility for some tech debt, I prefer for code to be agnostic to who works on it. At Google, I would suggest "TODO(bugnumber)" instead, as that would allow the bug to be assigned (or not) independent of the comment in the code.

Two alternatives to consider:

- `TODO(#NNN)`; file a bug capturing the work to be done. In that bug, describe when the work can start and possibly the target date.
- If possible, and I'm not at all confident it is possible, implement the future version and a compatibility fallback in a separate `.compat.py311` module that provides both the preferred implementation for Python 3.12+ and the version compatible with older Pythons. I use this pattern for non-typing compatibility and it works well because it provides a stark signal to any developer when it's no longer needed but also means that the preferred implementation is already tested and stable on later Pythons.

With that in mind, let me know how you want to proceed and we'll get this merged.

Reply:
That's clever. I will open a ticket for this!
About the other alternative, which would also be great, I'm afraid this time I can't implement future version because it involves new syntax, so it can't be mixed with older Pythons.

Reply:
@jaraco since this is a cross-project issue, I've filed https://github.com/coherent-oss/granary/issues/4.